kola/tests/podman/podman.go

```diff
@@ -0,0 +1,352 @@
+	image := "docker.io/library/nginx"
+	wwwRoot := "/usr/share/nginx/html"
+
+	dir := c.MustSSH(m, `mktemp -d`)
+
+	cmd := fmt.Sprintf("echo TEST PAGE > %s/index.html", string(dir))
+	c.MustSSH(m, cmd)
+
+	cmd = fmt.Sprintf("sudo podman run -d -p 80:80 -v %s/index.html:%s/index.html:z %s", string(dir), wwwRoot, image)
+	out := c.MustSSH(m, cmd)
+	id := string(out)
+
+	podIsRunning := func() error {
+		b, err := c.SSH(m, `curl -f http://localhost 2>/dev/null`)
+		if err != nil {
+			return err
+		}
+		if !bytes.Contains(b, []byte("TEST PAGE")) {
+			return fmt.Errorf("nginx pod is not running %s", b)
+		}
+		return nil
+	}
+
+	if err := util.Retry(10, 10*time.Second, podIsRunning); err != nil {
+		c.Fatal("Pod is not running")
+	}
+
+	// Test: Execute command in container
+	cmd = fmt.Sprintf("sudo podman exec %s echo hello", id[0:12])
+	out = c.MustSSH(m, cmd)
+
+	if string(out) != "hello" {
+		c.Fatal("Could not exec command in container")
+	}
+
+	// Test: Stop container
+	cmd = fmt.Sprintf("sudo podman stop %s", id[0:12])
+	c.MustSSH(m, cmd)
+	psInfo, err := getSimplifiedPsInfo(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	found := false
+	for _, container := range psInfo.containers {
+		if container.Id == id {
+			found = true
+			if container.Status != "exited" {
+				c.Fatalf("Container %s was not stopped. Current status: %s", id, container.Status)
+			}
+		}
+	}
+
+	if found == false {
+		c.Fatalf("Unable to find container %s in podman ps -a output", id)
+	}
+
+	// Test: Remove container
+	cmd = fmt.Sprintf("sudo podman rm %s", id[0:12])
+	c.MustSSH(m, cmd)
+	psInfo, err = getSimplifiedPsInfo(c, m)
+	found = false
+	for _, container := range psInfo.containers {
+		if container.Id == id {
+			found = true
+		}
+	}
+
+	if found == true {
+		c.Fatalf("Container %s should be removed. %v", id[0:12], psInfo.containers)
+	}
+
+	// Test: Delete container
+	cmd = fmt.Sprintf("sudo podman rmi %s", image)
+	out = c.MustSSH(m, cmd)
+	imageId := string(out)
+
+	cmd = fmt.Sprintf("sudo podman images | grep %s", imageId)
+	out, err = c.SSH(m, cmd)
+	if err == nil {
+		c.Fatalf("Image should be deleted but found %s", string(out))
+	}
+}
+
+// Test: Verify basic podman info information
+func podmanInfo(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	info, err := getPodmanInfo(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// test for known settings
+	expectedGraphDriver := "overlay"
+	if info.Store.GraphDriverName != expectedGraphDriver {
+		c.Errorf("Unexpected driver: %v != %v", expectedGraphDriver, info.Store.GraphDriverName)
+	}
+	expectedGraphRoot := "/var/lib/containers/storage"
+	if info.Store.GraphRoot != expectedGraphRoot {
+		c.Errorf("Unexected graph root: %v != %v", expectedGraphRoot, info.Store.GraphRoot)
+	}
+}
+
+// Test: Run podman with various options
+func podmanResources(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	genPodmanContainer(c, m, "sleep", []string{"sleep"})
```

Comment:
Why `sleep` over say `echo`?

Reply:
Sleep was used in the docker test but there's no particular reason we need it here.  `echo` works and its faster!  Thanks for the suggestion!